proxy/proxy.go

```diff
@@ -533,6 +539,26 @@ func callWeaveAttach(container *docker.Container, args []string) error {
 	return nil
 }
 
+func validateCIDRs(cidrs []string) error {
+	check := func(cidr string, prefixes []string) bool {
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(cidr, prefix) {
+				if _, err := address.ParseCIDR(strings.TrimPrefix(cidr, prefix)); err == nil {
+					return true
+				}
+			}
+		}
+		return false
+	}
```

Comment:
It would be clearer when lifted. Less indentation and it would be obvious that it doesn't close over anything.

Anyway, you could easily inline it. Which would be better.